cobalt/browser/browser_module.cc

```diff
@@ -1782,6 +1782,17 @@ void BrowserModule::FreezeInternal(SbTimeMonotonic timestamp) {
   // First freeze all our web modules which implies that they will release
   // their resource provider and all resources created through it.
   FOR_EACH_OBSERVER(LifecycleObserver, lifecycle_observers_, Freeze(timestamp));
+
+  if (network_module_) {
+    // Synchronously wait for storage to flush before returning from freezing.
+    base::WaitableEvent flush_finished = {
```

Comment:
This code is duplicated below. Can we put this function into some util to avoid copy-paste?

Reply:
Ah yeah, may as well put it in StorageManager as a synchronousflush.